### DISCUSSION TITLE
Getting speed and checkpoints passed

### DISCUSSION THREAD

**Discussion by Ahmed2060** [trackmania-rl/tmrl#50] [Q&A]
Is it possible that i can only get back speed of car and number of checkpoints passed from this library? as i am using my own scripts but this is what I am missing

Thank you

**Comment by yannbouteiller** [comment]
Sure, you can adapt [this function](https://github.com/trackmania-rl/tmrl/blob/b142b8ec95ef7ca59a89a23e6ec22795e842733d/tmrl/custom/custom_gym_interfaces.py#L193) to your application, this is where the speed and reward (i.e., number of checkpoints passed) are retrieved.

**Reply by Ahmed2060** [reply]
i mean like to take data from Tmrl_GrabData and feed it to my algorithm
